cpp/server/handler.h

```diff
@@ -32,21 +33,27 @@ class ThreadPool;
 class HttpHandler {
  public:
   // Does not take ownership of its parameters, which must outlive
-  // this instance. The "frontend" parameter can be NULL, in which
-  // case this server will not accept "add-chain" and "add-pre-chain"
-  // requests.
-  HttpHandler(JsonOutput* json_output, LogLookup<LoggedEntry>* log_lookup,
+  // this instance.
+  HttpHandler(LogLookup<LoggedEntry>* log_lookup,
               const ReadOnlyDatabase<LoggedEntry>* db,
               const ClusterStateController<LoggedEntry>* controller,
-              const CertChecker* cert_checker, Frontend* frontend,
-              Proxy* proxy, ThreadPool* pool, libevent::Base* event_base);
+              ThreadPool* pool, libevent::Base* event_base);
   ~HttpHandler();
 
   void Add(libevent::HttpServer* server);
 
- private:
+  void SetOutput(JsonOutput* json_output);
+  void SetProxy(Proxy* proxy);
+
+ protected:
+  // Implemented by subclasses which want to add their own extra http handlers.
+  virtual void AddHandlers(libevent::HttpServer* server) = 0;
```

Comment:
I could do, it's just that I figured that all subclasses will want to map the `get-entries`, `get-proof-by-hash`, `get-sth`, and `get-sth-consistency` handlers, so may as well just bake them in here.
If `Add` is virtual then the subclass has to either add them all itself, or include a call to the superclass `Add` implementation.  The former is a bunch of extra boilerplate for no real reason, and the latter would be ok but costs an extra line and could also lead to confusing "why are only *some* handlers missing" bugs.